Makefile

```diff
@@ -1,26 +1,53 @@
 
-NODE_ENV ?= development
-
 BIN := node_modules/.bin
-SRC := $(shell find lib -type f -name "*.js")
-
-BROWSERIFY_FLAGS :=  -t [ babelify --presets es2015 --plugins [ transform-react-jsx --pragma element ] ]
-DUO_FLAGS := --development
+SRC := $(shell find src -type f -name "*.js")
 
-clean:
-	rm -rf components build
+COVERAGE_BRANCH_THRESHOLD := 73
+COVERAGE_FUNCTION_THRESHOLD := 98
+COVERAGE_LINE_THRESHOLD := 98
+COVERAGE_STATEMENT_THRESHOLD := 98
 
-build/browserify.js: index.js $(SRC) build node_modules
-	$(BIN)/browserify $(BROWSERIFY_FLAGS) $< > $@
+lib: $(SRC) node_modules
+	$(BIN)/babel --out-dir $@ src
 
-build/duo.js: index.js $(SRC) build node_modules
-	$(BIN)/duo $(DUO_FLAGS) $< > $@
+clean:
+	rm -rf components lib coverage
+
+test: node_modules
+	$(BIN)/mochify \
+	  --phantomjs $(BIN)/phantomjs \
+	  --reporter spec \
+	  --transform babelify \
+	    test/index.js
+
+coverage: $(SRC) $(wildcard test/*.js) node_modules
+	$(BIN)/mochify \
+	  --phantomjs $(BIN)/phantomjs \
+	  --reporter spec \
+	  --transform babelify \
+	  --plugin [ \
+	    mochify-istanbul \
+	      --exclude '**/node_modules/**' \
+	      --exclude '**/test/**' \
+	      --report text \
+	      --report json \
+	      --dir $@ \
+	    ] \
+	   test/index.js
+
+check-coverage: coverage
+	$(BIN)/istanbul check-coverage \
+	  --branches $(COVERAGE_BRANCH_THRESHOLD) \
+	  --functions $(COVERAGE_FUNCTION_THRESHOLD) \
+	  --lines $(COVERAGE_LINE_THRESHOLD) \
+	  --statements $(COVERAGE_STATEMENT_THRESHOLD)
+
+build: lib node_modules
```

Comment:
rename to `check-build` or something, since `make build` is used everywhere for everything